### ISSUE TITLE
[Suggestion] Data from jsonable field into BackendList

### ISSUE THREAD

**Issue by Samorai** [octobercms/october#2990]
Hi to everyone!

##### Expected behavior
I have model with jsonable field named **foo** with values `{"bar":"baz"}`.
When I try to render it into Backend List, I'have wrote this code in columns.yml
```
foo.sss:
    type: text
```
And I wanted to see in list "baz".

##### Actual behavior
I had saw nothing

##### Resolving
I have added custom list column type named "array"
```
public function registerListColumnTypes()
    {
        return [
            'array' => function($value, $column, $record) {
                list($key, $values) = explode('.', $column->columnName);
                return array_get($record->$key, $values);
            }
        ];
    }
```
And change data into columns.yml file to 
```
foo.sss:
    type: array
```

##### Suggestion
I suggest to add evalArrayTypeValue to Backend\Widgets\Lists.

If it's OK, I can create PR with this feature.

**Comment by LukeTowers** [comment]
If this were to be added, the syntax for accessing a jsonable property should be the same as in `fields.yaml`: `property[key]`

**Comment by Samorai** [comment]
@LukeTowers 
The syntax for accessing jsonable field does not matters for me. 
But if you suggest the syntax `property[key]`, how you can access for lower properties like `property[key][key2][key3]` without errors if key does not exists?
Function array_get has resolve this problem.

**Comment by LukeTowers** [comment]
Well, either way you can resolve the issue right now by using a column type `partial` and accessing the value you wish to display directly through `<?= $record->property['key'] ?>`

**Comment by Samorai** [comment]
I know. I have resolved the issue using custom list column type.
But I don't really want to move this part of code between the projects. 
And I have suggest to discuss the syntax, what would I add the PR with resolving the issue globally for everyone.

**Comment by LukeTowers** [comment]
Just a couple side comments:

1. Support would not include sorting or searching support by default, anybody that wanted to utilize this on their columns would need to set `searchable: false` and `sortable: false` on the columns that utilized this to prevent the list controller's search or sort functionality from breaking.

2. I don't think that adding a new `type: array` is the correct way of solving this, since any type of data can be stored in a jsonable property, so a solution should implement it at the data retrieval level, not the rendering level.

**Comment by Samorai** [comment]
When I have jsonable value `{"foo":"foo_value", "bar": "bar_value"}` and I try to render with 
`type: text`. I have `foo_value, bar_value` in output.

**Comment by LukeTowers** [comment]
Not really sure what you mean by that @Samorai

**Comment by Samorai** [comment]
I'm talking about current behavior of rendering jsonable field if I try to use this code:
```
foo:
    type: text
```

`foo` is jsonable field. 

If it has value into DB: `{"bar":"baz", "bar_2", "baz_2"}` I see in the list `baz, baz_2`.

If it has value into DB: `{"bar", "baz", "foo"}` I see in the list `bar, baz, foo`.
@LukeTowers

**Comment by LukeTowers** [comment]
I'm suggesting that support should be added in the form of
`foo` jsonable property:
```json
{"date":"2017-01-01 20:10:32", "color": "#f2f2f2"}
```
YAML Column config:
```yaml
foo[date]:
    type: datetime

foo[color]
    type: colorpicker
```

And have that just work.

**Comment by LukeTowers** [comment]
Fixed by #2995
